e2e/test/csi_driver_test.go

```diff
@@ -61,33 +61,71 @@ var _ = Describe("Linode CSI Driver", func() {
 
 	Describe("Test", func() {
 		Context("Simple", func() {
+			var (
+				f    *framework.Invocation
+				size string
+				// storageClass = "linode-block-storage"
+				sts          *appsv1.StatefulSet
+				pod          *core.Pod
+				pvc          *core.PersistentVolumeClaim
+				file         = "/data/file.txt"
+				storageClass = "linode-block-storage"
+
+				writeFile = func(filename string) {
+					By("Writing a file into the Pod")
+					err := f.WriteFileIntoPod(filename, pod)
+					Expect(err).NotTo(HaveOccurred())
+				}
+
+				readFile = func(filename string) {
+					By("Checking if the created file is in the Pod")
+					err := f.CheckIfFileIsInPod(filename, pod)
+					Expect(err).NotTo(HaveOccurred())
+				}
+			)
 			Context("Block Storage", func() {
 				JustBeforeEach(func() {
 					By("Creating Persistent Volume Claim")
-					pvc = f.GetPersistentVolumeClaimObject(size, f.StorageClass, false)
-					err = f.CreatePersistentVolumeClaim(pvc)
+					pvc = f.GetPersistentVolumeClaimObject("test-pvc", f.Namespace(), size, storageClass, false)
+					err := f.CreatePersistentVolumeClaim(pvc)
 					Expect(err).NotTo(HaveOccurred())
 
 					By("Creating Pod with PVC")
-					pod = f.GetPodObject(podName1, pvc.Name)
+					pod = framework.GetPodObject("test-pod", f.Namespace(), pvc.Name)
 					err = f.CreatePod(pod)
 					Expect(err).NotTo(HaveOccurred())
 				})
 
 				AfterEach(func() {
 					By("Deleting the Pod with PVC")
-					err = f.DeletePod(pod.Name)
+					err := f.DeletePod(pod.ObjectMeta)
 					Expect(err).NotTo(HaveOccurred())
 
 					By("Waiting for the Volume to be Detached")
-					waitForOperation()
+					volumeID, err := f.GetVolumeID(pvc)
+
+					Eventually(func() bool {
+						isAttached, err := f.IsVolumeDetached(volumeID)
+						if err != nil {
+							return false
+						}
+						return isAttached
+					}, f.Timeout, f.RetryInterval).Should(BeTrue())
 
 					By("Deleting the PVC")
 					err = f.DeletePersistentVolumeClaim(pvc.ObjectMeta)
 					Expect(err).NotTo(HaveOccurred())
 
 					By("Waiting for the Volume to be Deleted")
-					waitForOperation()
+					Expect(err).NotTo(HaveOccurred())
+
+					Eventually(func() bool {
```

Comment:
This swallows the error, makes debugging close to impossible.

Reply:
all of them were written this way so i followed suite, can we agree to make an issue to rewrite these better and not include this as a part of getting these things semi functional again?

Reply:
Why can't we just write them properly in the first place? I only commented this one, because it looks like it's a new one, I'm not suggesting we redo all of them.

(The ticket to refactor these is already in place and I'm working on it - I got stopped by these tests being broken.)

Reply:
i dont think this is valid feedback any more after rewriting stuff, im going to resolve and you can open a new one against anything in the current pr